servicetalk-dns-discovery-netty/src/test/java/io/servicetalk/dns/discovery/netty/DefaultDnsClientTest.java

```diff
@@ -984,6 +984,56 @@ void srvExceptionInSubscriberOnNext() throws Exception {
         latchOnError.await();
     }
 
+    @Test
+    void capsMaxTTLForARecord() throws Exception {
+        setup(builder -> builder.maxTTL(3));
+        final String domain = "servicetalk.io";
+        String ip1 = nextIp();
+        String ip2 = nextIp();
+        recordStore.addIPv4Address(domain, 5, ip1);
+
+        TestPublisherSubscriber<ServiceDiscovererEvent<InetAddress>> subscriber = dnsQuery(domain);
+        Subscription subscription = subscriber.awaitSubscription();
+        subscription.request(Long.MAX_VALUE);
+
+        assertEvent(subscriber.takeOnNext(), ip1, AVAILABLE);
+        recordStore.removeIPv4Address(domain, 5, ip1);
+        recordStore.addIPv4Address(domain, 5, ip2);
+        advanceTime(3);
```

Comment:
Consider using a local variable to store `maxTTL` value to make it clear you advance only `maxTTL`.